sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGenerateExec.scala

```diff
@@ -826,13 +827,99 @@ class GpuGenerateIterator(
   // Need to ensure these are closed in case of failure.
   inputs.foreach(scb => use(scb))
 
-  // apply generation on each (sub)batch
-  private val retryIter = withRetry(inputs.toIterator, splitSpillableInHalfByRows) { attempt =>
-    withResource(attempt.getColumnarBatch()) { batch =>
-      generator.generate(batch, generatorOffset, outer)
+  def generateSplitSpillableInHalfByRows: SpillableColumnarBatch => Seq[SpillableColumnarBatch] = {
+    (spillable: SpillableColumnarBatch) => {
+      withResource(spillable) { _ =>
+        val toSplitRows = spillable.numRows()
+        if (toSplitRows == 1) {
+          // single row, we need to actually add row duplicates, then split
+          val tbl = withResource(spillable.getColumnarBatch()) { src =>
+            GpuColumnVector.from(src)
+          }
+          withResource(tbl) { _ =>
+            if (tbl.getColumn(generatorOffset).getNullCount == 1) {
+              // 1 row, and the element to explode is null, we cannot
+              // handle this case
+              throw new GpuSplitAndRetryOOM(
+                s"GPU OutOfMemory: cannot split a batch of 1 rows when " +
+                  s"the array to explode is null!")
+            }
+            // borrow the column to explode, in order to split it as a table
+            // explodedTbl is 1 row per element of the list
+            val explodedTbl =
+            withResource(new Table(tbl.getColumn(generatorOffset))) { generatorTmpTable =>
+              // we use regular `explode` because we know we have 1 list column
+              // and it isn't null => this operation will produce N rows, 1 row per element
+              // of the column at `generatorOffset`
+              generatorTmpTable.explode(0)
+            }
+
+            // Because we are likely to be here solving a cuDF column length limit
+            // we are just going to split in half blindly until we can fit it.
+            val splits = withResource(explodedTbl) { _ =>
+              if (explodedTbl.getRowCount < 2) {
+                // we should never hit this
+                throw new GpuSplitAndRetryOOM(
+                  s"GPU OutOfMemory: cannot split a batch of 1 rows when " +
+                    s"the array has a single element!")
+              }
+              val splitIx = Seq(explodedTbl.getRowCount.toInt / 2)
+              explodedTbl.contiguousSplit(splitIx: _*)
```

Comment:
Nit: Not sure why we're making a sequence-of-1 only to undo it later.
```suggestion
              val splitIx = explodedTbl.getRowCount.toInt / 2
              explodedTbl.contiguousSplit(splitIx)
```
or just:
```suggestion
              explodedTbl.contiguousSplit(explodedTbl.getRowCount.toInt / 2)
```